File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    void subsetSol(vector<vector<int>>& allsubset,vector<int>& nums, int count , vector<int>& subsets ){
        if(count == nums.size()){
            allsubset.push_back(subsets);
            return;
        }
        subsets.push_back(nums[count]);
        subsetSol(allsubset , nums , count + 1 , subsets);
        subsets.pop_back();
        subsetSol(allsubset , nums , count + 1, subsets);
        return;
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> allsubset;
        allsubset.reserve(1 << nums.size());
        vector<int> subsets = {};
        int count = 0;
        subsetSol(allsubset , nums , count , subsets);
        return allsubset;
    }
};

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        seen = [0] * len(nums)
        tim = 0
        result = []
        curr = []
        def back(tim , curr):
            if tim == len(nums):
                result.append(list(curr))
                return
            curr.append(nums[tim])
            back(tim + 1 , curr)
            curr.pop()
            back(tim + 1 , curr)
        back(0 , curr)
        return result
        